libindy/ci/ubuntu.dockerfile

```diff
@@ -38,13 +38,16 @@ RUN pip3 install -U \
 	plumbum \
 	deb-pkg-tools
 
+ARG LIBSODIUM_VERSION=1.0.17
+
 RUN cd /tmp && \
-   curl https://download.libsodium.org/libsodium/releases/old/libsodium-1.0.14.tar.gz | tar -xz && \
-    cd /tmp/libsodium-1.0.14 && \
-    ./configure --disable-shared && \
+   curl https://download.libsodium.org/libsodium/releases/libsodium-$LIBSODIUM_VERSION.tar.gz | tar -xz && \
+    cd /tmp/libsodium-$LIBSODIUM_VERSION && \
+    ./configure && \
```

Comment:
@AxelNennker could you share the reason, why `--disable-shared` flag is removed in the PR?

Reply:
It did not built with this flag. 
https://build.sovrin.org/blue/organizations/jenkins/indy-sdk%2Findy-sdk-ci/detail/PR-1585/4/pipeline/
I did not attempt to find the reason because the static library is built nevertheless and linked  because of `cargo:rustc-link-lib=static=sodium` 
https://github.com/hyperledger/indy-sdk/blob/master/libindy/build.rs#L61

Reply:
One additional remark to 'it did not built'. It builds on my Ubuntu 16.04 with this flag but did not built on the CI server. Hard to debug for me. So I took the easy way. Hope that is ok.

Reply:
Thanks for clarification. It sounds very strange as in fact libindy build flow should use only static artifact of libsodium (as you mentioned above). May be it hide some other problem, so I will try to reproduce it in my environment and dockers to understand the reason.

Reply:
I just built libsodium-1.0.17 again.
```curl https://download.libsodium.org/libsodium/releases/libsodium-1.0.17.tar.gz | tar xz
cd libsodium-1.0.17/
./configure --disable-shared && make
cd src/libsodium/.libs
ar -x libsodium.a
readelf --relocs librdrand_la-randombytes_salsa20_random.o | egrep '(GOT|PLT|JU?MP_SLOT)'00000000001d  002900000004 R_X86_64_PLT32    0000000000000000 close - 4
000000000052  002a00000004 R_X86_64_PLT32    0000000000000000 sodium_memzero - 4
0000000000bf  002b00000004 R_X86_64_PLT32    0000000000000000 syscall - 4
0000000000c8  002c00000004 R_X86_64_PLT32    0000000000000000 __errno_location - 4
000000000124  002d00000004 R_X86_64_PLT32    0000000000000000 __assert_fail - 4
000000000138  002e00000004 R_X86_64_PLT32    0000000000000000 getpid - 4
000000000143  002f00000004 R_X86_64_PLT32    0000000000000000 sodium_misuse - 4
00000000017a  003000000004 R_X86_64_PLT32    0000000000000000 gettimeofday - 4
0000000001e4  002e00000004 R_X86_64_PLT32    0000000000000000 getpid - 4
000000000222  003100000004 R_X86_64_PLT32    0000000000000000 read - 4
000000000281  002c00000004 R_X86_64_PLT32    0000000000000000 __errno_location - 4
000000000292  002f00000004 R_X86_64_PLT32    0000000000000000 sodium_misuse - 4
0000000002c1  002c00000004 R_X86_64_PLT32    0000000000000000 __errno_location - 4
0000000002cc  003200000004 R_X86_64_PLT32    0000000000000000 sodium_runtime_has_rdr - 4
000000000329  003300000004 R_X86_64_PLT32    0000000000000000 open - 4
00000000035b  003400000004 R_X86_64_PLT32    0000000000000000 poll - 4
000000000373  002900000004 R_X86_64_PLT32    0000000000000000 close - 4
000000000395  002900000004 R_X86_64_PLT32    0000000000000000 close - 4
0000000003c3  003500000004 R_X86_64_PLT32    0000000000000000 __fxstat - 4
0000000003de  002900000004 R_X86_64_PLT32    0000000000000000 close - 4
0000000003f4  003300000004 R_X86_64_PLT32    0000000000000000 open - 4
000000000411  003600000004 R_X86_64_PLT32    0000000000000000 fcntl - 4
000000000424  003600000004 R_X86_64_PLT32    0000000000000000 fcntl - 4
00000000046d  002d00000004 R_X86_64_PLT32    0000000000000000 __assert_fail - 4
000000000472  003700000004 R_X86_64_PLT32    0000000000000000 __stack_chk_fail - 4
0000000004df  003800000004 R_X86_64_PLT32    0000000000000000 crypto_stream_salsa20 - 4
0000000005d2  002d00000004 R_X86_64_PLT32    0000000000000000 __assert_fail - 4
000000000629  003800000004 R_X86_64_PLT32    0000000000000000 crypto_stream_salsa20 - 4
0000000006a2  003900000004 R_X86_64_PLT32    0000000000000000 crypto_stream_salsa20_ - 4
0000000006d5  002d00000004 R_X86_64_PLT32    0000000000000000 __assert_fail - 4
```
which looks like it should, I think. And CFLAGS has `-fPIC -fPIE` so that looks good as well.

Reply:
Maybe there is another version of libsodium.a installed that is found and used and that triggers the error...

Reply:
Any preference on how to proceed?:
1.  just take what Ubuntu offers and stay with the old version, that is give up here or
2.  set LIBSODIUM_DIR to the fresh libsodium so that Rust finds the new one
3.  compile zmq from source as well and don't install libzmq3-dev

I prefer option 3 and I'll give it a try

Reply:
Let's try option 3 until we migrate to ubuntu 18.04